modules/Core/src/main/java/org/terasology/logic/inventory/CharacterInventorySystem.java

```diff
@@ -115,6 +115,7 @@ public void onDropItemRequest(DropItemRequest event, EntityRef character) {
             }
         }
 
+        pickupItem.getComponent(ItemComponent.class).pickUpTime = 2000;
```

Comment:
About this, the idea was for the item to drop from the player's hand for that set amount of time. If I put that in the onDropItemEvent, I'd have to either have a way to check what dropped it to make sure I set the right amount of time. (whereas the DropItemRequest comes directly from the character, thus I control the time to pick up from it being thrown by the character instead of every block)

Having the pickUpTime dependant on the item entity is a bit different mechanic, but I could just do that if you think it's a better idea.